From cef04cdcb4275592ed6b6822e1939de44acce15b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micka=C3=ABl=20Sala=C3=BCn?= <mic@digikod.net>
Date: Fri, 26 May 2017 20:44:00 +0200
Subject: [PATCH 07/41] Documentation/dev-tools: Use reStructuredText markups
 for kselftest
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Include and convert kselftest to the Sphinx format.

Signed-off-by: Mickaël Salaün <mic@digikod.net>
Acked-by: Kees Cook <keescook@chromium.org>
Cc: Jonathan Corbet <corbet@lwn.net>
Cc: Shuah Khan <shuah@kernel.org>
Signed-off-by: Shuah Khan <shuahkh@osg.samsung.com>
---
 Documentation/dev-tools/index.rst     |  1 +
 Documentation/dev-tools/kselftest.rst | 67 +++++++++++++++++++++--------------
 2 files changed, 41 insertions(+), 27 deletions(-)

diff --git a/Documentation/dev-tools/index.rst b/Documentation/dev-tools/index.rst
index 07d881147ef3..e50054c6aeaa 100644
--- a/Documentation/dev-tools/index.rst
+++ b/Documentation/dev-tools/index.rst
@@ -23,6 +23,7 @@ whole; patches welcome!
    kmemleak
    kmemcheck
    gdb-kernel-debugging
+   kselftest
 
 
 .. only::  subproject and html
diff --git a/Documentation/dev-tools/kselftest.rst b/Documentation/dev-tools/kselftest.rst
index 5bd590335839..9232ce94612c 100644
--- a/Documentation/dev-tools/kselftest.rst
+++ b/Documentation/dev-tools/kselftest.rst
@@ -1,4 +1,6 @@
+======================
 Linux Kernel Selftests
+======================
 
 The kernel contains a set of "self tests" under the tools/testing/selftests/
 directory. These are intended to be small tests to exercise individual code
@@ -15,29 +17,34 @@ hotplug test is run on 2% of hotplug capable memory instead of 10%.
 Running the selftests (hotplug tests are run in limited mode)
 =============================================================
 
-To build the tests:
-  $ make -C tools/testing/selftests
+To build the tests::
+
+    make -C tools/testing/selftests
+
+To run the tests::
 
+    make -C tools/testing/selftests run_tests
 
-To run the tests:
-  $ make -C tools/testing/selftests run_tests
+To build and run the tests with a single command, use::
 
-To build and run the tests with a single command, use:
-  $ make kselftest
+    make kselftest
 
-- note that some tests will require root privileges.
+Note that some tests will require root privileges.
 
 
 Running a subset of selftests
-========================================
+=============================
+
 You can use the "TARGETS" variable on the make command line to specify
 single test to run, or a list of tests to run.
 
-To run only tests targeted for a single subsystem:
-  $  make -C tools/testing/selftests TARGETS=ptrace run_tests
+To run only tests targeted for a single subsystem::
+
+    make -C tools/testing/selftests TARGETS=ptrace run_tests
 
-You can specify multiple tests to build and run:
-  $  make TARGETS="size timers" kselftest
+You can specify multiple tests to build and run::
+
+    make TARGETS="size timers" kselftest
 
 See the top-level tools/testing/selftests/Makefile for the list of all
 possible targets.
@@ -46,13 +53,15 @@ possible targets.
 Running the full range hotplug selftests
 ========================================
 
-To build the hotplug tests:
-  $ make -C tools/testing/selftests hotplug
+To build the hotplug tests::
+
+    make -C tools/testing/selftests hotplug
+
+To run the hotplug tests::
 
-To run the hotplug tests:
-  $ make -C tools/testing/selftests run_hotplug
+    make -C tools/testing/selftests run_hotplug
 
-- note that some tests will require root privileges.
+Note that some tests will require root privileges.
 
 
 Install selftests
@@ -62,13 +71,15 @@ You can use kselftest_install.sh tool installs selftests in default
 location which is tools/testing/selftests/kselftest or a user specified
 location.
 
-To install selftests in default location:
-   $ cd tools/testing/selftests
-   $ ./kselftest_install.sh
+To install selftests in default location::
 
-To install selftests in a user specified location:
-   $ cd tools/testing/selftests
-   $ ./kselftest_install.sh install_dir
+    cd tools/testing/selftests
+    ./kselftest_install.sh
+
+To install selftests in a user specified location::
+
+    cd tools/testing/selftests
+    ./kselftest_install.sh install_dir
 
 Running installed selftests
 ===========================
@@ -79,8 +90,10 @@ named "run_kselftest.sh" to run the tests.
 You can simply do the following to run the installed Kselftests. Please
 note some tests will require root privileges.
 
-cd kselftest
-./run_kselftest.sh
+::
+
+    cd kselftest
+    ./run_kselftest.sh
 
 Contributing new tests
 ======================
@@ -96,8 +109,8 @@ In general, the rules for selftests are
  * Don't cause the top-level "make run_tests" to fail if your feature is
    unconfigured.
 
-Contributing new tests(details)
-===============================
+Contributing new tests (details)
+================================
 
  * Use TEST_GEN_XXX if such binaries or files are generated during
    compiling.
-- 
2.13.0

